### DISCUSSION TITLE
Best way to learn Langium

### DISCUSSION THREAD

**Discussion by paulfrench** [eclipse-langium/langium#358] [General]
Hello all, I'm hoping people will post their recommendations here on what to read to get started etc.

For example the Langium grammar is similar to XText grammar but there are significant differences. Is this documented anywhere?

Personally I'm struggling to get to grips with the grammar in the first place.

When do you try and put in stricter grammar for better auto-completion or instead use validation to enforce some grammar rules?

Useful links I am aware of so far...

https://langium.org/
https://www.typefox.io/blog/langium-the-new-language-engineering-tool
https://www.eclipse.org/Xtext/documentation/index.html

**Comment by spoenemann** [comment]
Hi @paulfrench! We will have a documentation of the Langium grammar at https://langium.org/docs/ soon. Other parts of the framework are going to be documented as well.

Another way to learn is to look at examples. Apart from those included in the repository (https://github.com/langium/langium/tree/main/examples), we are working on a larger example, which is still work in progress: https://github.com/TypeFox/langium-ui-framework

**Comment by paulfrench** [comment]
Another option I found very useful. Sign up to https://learning.oreilly.com/ for 10 day free trial. Then read the book...

**Implementing Domain-Specific Languages with Xtext and Xtend - Second Edition**

**Comment by goto40** [comment]
A quick introduction has been presented by Gabriele Tomassetti: https://tomassetti.me/quickly-create-dsls-with-langium/
